### PR TITLE
Addition of valid cudaMemcpyKind to cudaMemcpy

### PR DESCRIPTION
**Pull request by Pinkyboi**
For now in `copy_CudaArray_Data` the cudaMemcpyKind field of cudaMemcpy is hardcoded, this PR aims to make it so that field can be deduced from the variable's memory location and the current context of the function.

### CODE REVIEW

File: pyccel/codegen/printing/ccudacode.py [hunk 1]
```diff
@@ -519,19 +519,23 @@ def copy_CudaArray_Data(self, expr):
         declare_dtype = self.find_in_dtype_registry(self._print(rhs.dtype), rhs.precision)
         dtype = self.find_in_ndarray_type_registry(self._print(rhs.dtype), rhs.precision)
         arg = rhs.arg if isinstance(rhs, (CudaArray, CupyArray)) else rhs
+        if rhs.current_location == 'device' and rhs.memory_location == 'host':
+            errors.report("Cannot copy data from device to host", symbol=expr, severity='error')
+        memory_location = 'Host' if rhs.memory_location == 'host' else 'Device'
```
Comment (bauom) [inline pyccel/codegen/printing/ccudacode.py:524]: does this impact `managed` arrays ?

File: pyccel/codegen/printing/ccudacode.py [hunk 2]
```diff
@@ -519,19 +519,23 @@ def copy_CudaArray_Data(self, expr):
         declare_dtype = self.find_in_dtype_registry(self._print(rhs.dtype), rhs.precision)
         dtype = self.find_in_ndarray_type_registry(self._print(rhs.dtype), rhs.precision)
         arg = rhs.arg if isinstance(rhs, (CudaArray, CupyArray)) else rhs
+        if rhs.current_location == 'device' and rhs.memory_location == 'host':
+            errors.report("Cannot copy data from device to host", symbol=expr, severity='error')
```
Comment (bauom) [inline pyccel/codegen/printing/ccudacode.py:523]: you can copy data from `device` to `host`.

the problem is when creating a `host` array from the `device`.

Comment (Pinkyboi) [reply pyccel/codegen/printing/ccudacode.py:523]: Roger that i'll change that error message.

File: pyccel/codegen/printing/ccudacode.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Pinkyboi) [reply pyccel/codegen/printing/ccudacode.py:524]: Could you be more precise please i don't where this portion of the code could cause a problem

[the remark this answers is not in the document]
Comment (bauom) [reply pyccel/codegen/printing/ccudacode.py:524]: ah okey i see that the variable `memory_location` is used just to construct the `memcpy_kind`
could you change the name to something like `memcpy_kind_src` maybe ?

[the remark this answers is not in the document]
Comment (Pinkyboi) [reply pyccel/codegen/printing/ccudacode.py]: I guess something like `memcpy_kind_dest` is more suited for this case i'll change that.

File: pyccel/codegen/printing/ccudacode.py (hunk 2 above)
Comment (Pinkyboi) [reply pyccel/codegen/printing/ccudacode.py]: Fixed in [aa5e773](https://github.com/pyccel/pyccel/pull/1228/commits/aa5e773c542e0ecc6f193ac775b43dfde1d35b16).

File: pyccel/codegen/printing/ccudacode.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Pinkyboi) [reply pyccel/codegen/printing/ccudacode.py]: Fixed in [e3e13f1](https://github.com/pyccel/pyccel/pull/1228/commits/e3e13f1053e123c5288d81de2b2523ad13135a73).

**Review by bauom — APPROVED** [review]

**Pull request merged**